torch/serialization.py

```diff
@@ -1954,6 +1954,7 @@ def load_tensor(dtype, numel, key, location):
         if torch._guards.detect_fake_mode(None) is not None:
             nbytes = numel * torch._utils._element_size(dtype)
             storage = torch.UntypedStorage(nbytes, device="meta")
+            storage._checkpoint_offset = zip_file.get_record_offset(name)
```

Comment:
Should we use the same logic as below to avoid reading the zip when we can for best perf?